libpod/runtime_ctr.go

```diff
@@ -460,6 +460,11 @@ func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (_ *Contai
 	} else if err := r.state.AddContainer(ctr); err != nil {
 		return nil, err
 	}
+
+	if err := ctr.prepare(); err != nil {
```

Comment:
As discussed before, this *cannot* be the default. There will be significant performance implications from it.

Also, this must be following by a `ctr.cleanup()` to ensure we do not leak mountpoints and network configurations.

Reply:
I will add a `defer ctr.cleanup()` statement. After checking the `prepare()` code I thought calling it again will be a no-op whether it is invoked using `podman container init` or `podman run`.
For my knowledge, how will be create performance implications?

Reply:
This adds another mount and unmount to the early-init code of Podman, plus a setup and teardown of the full CNI network stack. These are the two most expensive operations during container creation, and forcing the `podman run` command to do them twice is going to significantly increase runtime in typical cases.

The core issue here is that not all calls to container creation are `podman run`. It is perfectly legal for someone to `podman create` a container, wait a long time (years, even), and then `podman start` said container. We cannot keep the container's mounts and network namespace alive during this time - the end result of container creation must be a container with no resources consumed. This is why we must clean up after the initial mount of the container when doing this copy-up, and why it cannot be the default.

Reply:
> the end result of container creation must be a container with no resources consumed

Thank you @mheon . Agree with this.